windows.js

```diff
@@ -126,9 +186,22 @@ async function setupMSWin() {
 /* Sets MSVC environment for use in Actions
  *   allows steps to run without running vcvars*.bat, also for PowerShell
  *   adds a convenience VCVARS environment variable
- *   this assumes a single Visual Studio version being available in the windows-latest image */
+ *   this assumes a single Visual Studio version being available in the Windows images */
 export function addVCVARSEnv() {
-  const vcVars = '"C:\\Program Files (x86)\\Microsoft Visual Studio\\2019\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"'
+  let vcVars = ''
+  switch (virtualEnv) {
+    case 'windows-2016':
+      vcVars = '"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\VC\\Auxiliary\\Build\\vcvars64.bat"'
```

Comment:
So before it looks like we were using MSVC 2019 on windows-2016, and now we'll use MSCV 2017 there. Any risk of incompatibility?

Reply:
> Any risk of incompatibility?

Actually, it's always been broken, VS 2019 isn't installed on windows-2016.

We don't test it here (mswin & windows-2016).  On Actions, ruby/ruby only builds mswin with windows-2019.  And, the only publicly available built, ruby-loco's mswin, is built with 2019.

Lastly, I think GitHub is removing the windows-2016 image in March?  So, leave it an remove when the image is removed?

Reply:
The change is fine, thanks for the details.